cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -567,6 +534,44 @@ void setReplicaLoad(TopicPartition tp, AggregatedMetricValues aggregatedMetricVa
     _load.addMetricValues(aggregatedMetricValues, windows);
   }
 
+  /**
+   * Populate the information of replica placement over disk for the broker.
+   *
+   * @param logdir The logdir of the disk.
+   * @param replicaInfos the replicas hosted by this disk.
+   */
+  public void populateDiskInfo(String logdir, Map<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> replicaInfos) {
+    Disk disk = disk(logdir);
+    replicaInfos.forEach((key, value) -> {
+      try {
+        if (!value.isFuture) {
+          disk.addReplica(replica(key), true);
+        }
+      } catch (NullPointerException e) {
+        // Let it go.
```

Comment:
Based on our offline discussion, can we make sure that if the user intends to `populateReplicaPlacementInfo`, then we always throw an exception unless we have the complete logDir capacity information to avoid making an intra-broker replica move optimization with missing disk capacity info?

Reply:
The exception will be thrown if the replica is online but cannot get the disk correspond to its logdir(which means a real miss from BrokerCapacityResolver).

There is another case, if the disk is dead, we expect that BrokerCapacityResolver does not report that disk information, and the replica status in this case will be offline.

Reply:
Looks like this comment is no longer relevant? -- i.e. we removed the function.